torch/distributed/rpc/__init__.py

```diff
@@ -71,7 +76,38 @@ def init_rpc(
                 are available.
         """
 
-        if not rpc_backend_options:
+        # To avoid breaking users that passed a ProcessGroupRpcBackendOptions
+        # without specifying the backend as PROCESS_GROUP when that was the
+        # default, we try to detect the backend from the options when only the
+        # latter is passed.
+        if backend is None and rpc_backend_options is not None:
+            for candidate_backend in BackendType:
+                if isinstance(
+                    rpc_backend_options,
+                    type(
+                        backend_registry.construct_rpc_backend_options(
+                            candidate_backend
+                        )
+                    ),
+                ):
+                    backend = candidate_backend
+                    break
```

Comment:
Are these codecov warnings false positives? It does seem like your test should cover this line?

Reply:
That's unclear. I'm assuming that the coverage is collected from this job: https://app.circleci.com/pipelines/github/pytorch/pytorch/217281/workflows/3cfc25f2-225c-4be1-9ad5-a7942dc829e7/jobs/7659664 (since it's the only one with `coverage` in the name). And I checked the logs and the tests are run.

One option is that the odd on-the-fly class construction trick we're doing to assemble the RPC test suites makes it hard for the coverage tool to map the test back to its original lines.

Or I misunderstood how this works.